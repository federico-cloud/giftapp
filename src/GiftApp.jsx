import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import PropTypes from 'prop-types';
import { GifGrid } from './components/GifGrid';

export const GiftApp = () => {

    const [categorias, setCategorias] = useState([]);

  return (
    <>
   
        <h2>GiftApp</h2>
        <AddCategory setCategorias={setCategorias}/>
        <hr/>

        <ol>
            {
                categorias.map( categoria => ( 
                    <GifGrid 
                        key = {categoria}
                        categoria = {categoria}
                    />
                ))
            }
        </ol>

    </>
  )
}

AddCategory.propTypes = {
    setCategorias: PropTypes.func.isRequired
}
