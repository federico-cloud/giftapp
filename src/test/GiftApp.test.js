import React from 'react';
import { shallow } from 'enzyme';
import { GiftApp } from '../GiftApp';

describe('Pruebas en <GiftApp />', () => {

    test('Debe mostrar el componente correctamente', () => {

        const wrapper = shallow(<GiftApp />);

        expect(wrapper).toMatchSnapshot();

    })

    test('Debe mostrar una lista de categorias', () => {

        const categorias = ['Saitama', 'Goku'];

        const wrapper = shallow(<GiftApp defaultCategorias={categorias} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categorias.length);

    })

        
})