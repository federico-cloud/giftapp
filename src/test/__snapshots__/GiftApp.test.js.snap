// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GiftApp /> Debe mostrar el componente correctamente 1`] = `
<Fragment>
  <h2>
    GiftApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en <GiftApp /> Debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GiftApp
  </h2>
  <AddCategory
    setCategorias={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      categoria="Saitama"
      key="Saitama"
    />
    <GifGrid
      categoria="Goku"
      key="Goku"
    />
  </ol>
</Fragment>
`;
